packages/core-data/src/resolvers.js

```diff
@@ -279,6 +289,44 @@ export const getEntityRecords =
 						response.headers.get( 'X-WP-TotalPages' )
 					),
 				};
+			} else if (
+				query.per_page === -1 &&
+				query[ RECEIVE_INTERMEDIATE_RESULTS ] === true
```

Comment:
I actually can see this as being useful as a public API.  That said, I wonder about the impact on the "resolution state" and things like that.

The other thing I'm wondering, is that basically this means that the "pagination" logic for "-1" has moved within the resolver, so there's a potential to actually remove the code that is responsible for this in `apiFetch` and just reuse the same code here. The only different being that the action to "fill" the state won't be call until the end if `RECEIVE_INTERMEDIATE_RESULTS` is false.

Reply:
We could. I also saw that the resolver is anyway already dependent on knowledge of WP Total Pages etc., seen in trunk already 20 lines below. Do you think we should do it here or try separately?

Regarding the resolution state, I guess that's one thing to talk about before making this public. I think it's fine to resolve only after all requests are completed. It does not prevent components from selecting the unresolved state.

Reply:
> Do you think we should do it here or try separately?

I'd say separately.

Reply:
I don't mind separate but I feel like we can follow-up on this rather than leave it as nice to improve later.

Reply:
@youknowriad Well, can we remove it from api-fetch though? We'd break back compat? Not all -1 requests necessarily go through this resolver.

Reply:
Good point :) maybe we can deprecate, WDYT?

Reply:
We could I guess. I'll look into it tomorrow. In the meantime, I'll merge this PR so the performance problem is addressed.